src/gui/src/LogWindow.cpp

```diff
@@ -32,33 +34,58 @@ LogWindow::LogWindow(QWidget *parent) :
     // repeatedly until InputLeap is finished
     setAttribute(Qt::WA_DeleteOnClose, false);
     setupUi(this);
+    
+    // Use a timer to flush the buffer every 100 milliseconds
+    QTimer* timer = new QTimer(this);
+    connect(timer, &QTimer::timeout, this, &LogWindow::flushBuffer);
+    timer->start(100);
 }
 
 void LogWindow::startNewInstance()
 {
     // put a space between last log output and new instance.
-    if (!m_pLogOutput->toPlainText().isEmpty())
-        appendRaw("");
+    if (!buffer.isEmpty())
+        buffer += '\n';
 }
 
 void LogWindow::appendInfo(const QString& text)
 {
-    appendRaw(getTimeStamp() + " INFO: " + text);
+    buffer += getTimeStamp() + " INFO: " + text + '\n';
```

Comment:
Dont use concatantation insead do this
create a private for the widget
 `QString m_infoLine = QStringLiteral("%1 INFO: %2 \n");`
use this in place of your string 
`buffer.append(m_infoLine.arg(getTimeStamp(), text));`

using concatanation is very slow each + will cause a new check and resize of the string for the sting on the right side of the + for the string here it happens 3 times then again for the buffer... using append and/or arg subs will make this once for the string and once for the buffer.

Making our line a QStringLiteral will reduse the memory use as the template string is made at build time and stored in the binary using it at runtime costs nothing.